firedrake/utility_meshes.py

```diff
@@ -572,3 +571,192 @@ def UnitIcosahedralSphereMesh(refinement_level=0, reorder=None):
+    r = (a**2 + lX**2 + lY**2)**0.5
+
+    # Now we need to compute the gnonomic transformation
+    # for each of the panels
+    panel_numbering.shape = (6, Nx**2)
+
+    def coordinates_on_panel(panel_num, X, Y, Z):
+        I = panel_numbering[panel_num, :]
+        coords[I, 0] = radius / r * X
+        coords[I, 1] = radius / r * Y
+        coords[I, 2] = radius / r * Z
+
+    coordinates_on_panel(0, lX, lY, -a)
+    coordinates_on_panel(1, -a, lY, -lX)
+    coordinates_on_panel(2, a, lY, lX)
+    coordinates_on_panel(3, lX, a, lY)
+    coordinates_on_panel(4, lX, -a, -lY)
+    coordinates_on_panel(5, lX, -lY, a)
+
+    # Now we need to build the face numbering
+    # in local coordinates
+    vertex_numbers = np.arange(Nx**2).reshape(Nx, Nx)
+    local_faces = np.zeros(((Nx-1)**2, 4), dtype=np.int32)
+    local_faces[:, 0] = vertex_numbers[:-1, :-1].reshape(-1)
+    local_faces[:, 1] = vertex_numbers[1:, :-1].reshape(-1)
+    local_faces[:, 2] = vertex_numbers[1:, 1:].reshape(-1)
+    local_faces[:, 3] = vertex_numbers[:-1, 1:].reshape(-1)
+
+    cells = panel_numbering[:, local_faces].reshape(-1, 4)
+    return cells, coords
+
+
+def CubedSphereMesh(radius, refinement_level=0, reorder=None,
+                    use_dmplex_refinement=False):
+    """Generate an cubed approximation to the surface of the
+    sphere.
+
+    :arg radius: The radius of the sphere to approximate.
+    :kwarg refinement_level: optional number of refinements (0 is a cube).
+    :kwarg reorder: (optional), should the mesh be reordered?
+    :kwarg use_dmplex_refinement: (optional), use dmplex to apply
+    the refinement.
+    """
+    if use_dmplex_refinement:
+        # vertices of a cube with an edge length of 2
+        vertices = np.array([[-1., -1., -1.],
+                             [1., -1., -1.],
+                             [-1., 1., -1.],
+                             [1., 1., -1.],
+                             [-1., -1., 1.],
+                             [1., -1., 1.],
+                             [-1., 1., 1.],
+                             [1., 1., 1.]])
+        # faces of the base cube
+        # bottom face viewed from above
+        # 2 3
+        # 0 1
+        # top face viewed from above
+        # 6 7
+        # 4 5
+        faces = np.array([[0, 1, 3, 2],  # bottom
+                          [4, 5, 7, 6],  # top
+                          [0, 1, 5, 4],
+                          [2, 3, 7, 6],
+                          [0, 2, 6, 4],
+                          [1, 3, 7, 5]], dtype=np.int32)
+
+        plex = _from_cell_list(2, faces, vertices)
+        plex.setRefinementUniform(True)
+        for i in range(refinement_level):
+            plex = plex.refine()
+
+        # rescale points to the sphere
+        # this is not the same as the gnonomic transformation
+        coords = plex.getCoordinatesLocal().array.reshape(-1, 3)
+        scale = (radius / np.linalg.norm(coords, axis=1)).reshape(-1, 1)
+        coords *= scale
```

Comment:
Given that the DMPlex refinement method of generating the cubed sphere doesn't actually produce the same mesh (in terms of locations of the vertices), I'm inclined just to drop this option completely.